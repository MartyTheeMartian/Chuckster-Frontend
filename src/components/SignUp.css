.modal-close-btn {
  float: right;
}
