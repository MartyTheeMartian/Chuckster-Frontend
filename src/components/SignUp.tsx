import { useState } from 'react';
import Modal from 'react-modal';
import setModal from 'config/modal';
import './SignUp.css';


setModal(Modal);

export default function SignUp(): JSX.Element {

  const [modalVisible, setModalVisible] = useState(false);

  const displayModal = (): void => setModalVisible(true);
  const closeModal = (): void => setModalVisible(false);

  return (
    <div>
      <button
        type='button'
        onClick={displayModal}>
        Sign Up Here!
      </button>
      <Modal
        isOpen={modalVisible}
        onRequestClose={closeModal}
        contentLabel='Example Modal'>
        <button
          className='modal-close-btn'
          type='button'
          onClick={closeModal}>
          X
        </button>
        <h2>
          Sign Up
        </h2>
        <form>
          <input />
        </form>
      </Modal>
    </div>
  );
}
