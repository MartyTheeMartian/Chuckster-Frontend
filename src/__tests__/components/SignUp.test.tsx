import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignUp from 'components/SignUp';

it('renders App component', () => {
  const result = shallow(<SignUp />);
  expect(toJson(result)).toMatchSnapshot();
});
