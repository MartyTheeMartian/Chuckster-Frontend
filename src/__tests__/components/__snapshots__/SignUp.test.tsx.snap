// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component 1`] = `
<div>
  <button
    onClick={[Function]}
    type="button"
  >
    Sign Up Here!
  </button>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentElement={[Function]}
    contentLabel="Example Modal"
    isOpen={false}
    onRequestClose={[Function]}
    overlayElement={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    preventScroll={false}
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <button
      className="modal-close-btn"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
    <h2>
      Sign Up
    </h2>
    <form>
      <input />
    </form>
  </Modal>
</div>
`;
